### PR TITLE
create events when a product is added/removed from wishlist - ECI-177

### PR DESCRIPTION
**Pull request by tdieterich-drip**
Addresses [ECI-177](https://dripcom.atlassian.net/browse/ECI-177)
created tests around wish list modification
fixed helper calls to use the correct helper

Adding, removing via quantity, and removing via the trashcan all generate an event as expected.

### CODE REVIEW

File: app/code/community/Drip/Connect/Model/Observer/Wishlist/RemoveProductsWithoutQuantity.php [hunk 1]
```diff
@@ -21,22 +21,19 @@ protected function executeWhenEnabled($observer)
             return;
         }
 
-        //loop through each product and check quantity
-        if (filter_input_array(INPUT_POST) &&
-            (null !== filter_input(INPUT_POST, 'description')) &&
-            is_array(filter_input(INPUT_POST, 'description'))) {
-            foreach (filter_input(INPUT_POST, 'description') as $itemId => $description) {
+        $post_description = filter_input(INPUT_POST, 'description', FILTER_DEFAULT, FILTER_REQUIRE_ARRAY);
```
Comment (will-in-wi) [inline app/code/community/Drip/Connect/Model/Observer/Wishlist/RemoveProductsWithoutQuantity.php]: Nit: Magento standard is camel case
```suggestion
        $postDescription = filter_input(INPUT_POST, 'description', FILTER_DEFAULT, FILTER_REQUIRE_ARRAY);
```

File: devtools_m1/cypress/integration/Wishlists/steps.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import { Given, When, Then } from "cypress-cucumber-preprocessor/steps"
+import { mockServerClient } from "mockserver-client"
+import { getCurrentFrontendDomain, getCurrentFrontendWebsiteId } from "../../lib/frontend_context"
+
+const Mockclient = mockServerClient("localhost", 1080);
+
+When('I create an account', function() {
```
Comment (will-in-wi) [inline devtools_m1/cypress/integration/Wishlists/steps.js]: Might be worth extracting this and the one in `devtools_m1/cypress/integration/CustomerCartInteractions/steps.js` into the shared steps.

File: devtools_m1/cypress/integration/Wishlists/steps.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    case 'grouped':
+      cy.get('#product_addtocart_form input[name="super_group[2]"]').clear().type('1')
+      cy.get('#product_addtocart_form input[name="super_group[3]"]').clear().type('1')
+      break;
+    case 'simple':
+    case 'bundle': // For now, we only have one option for each bundle option, so we don't have to do anything.
+      // Do nothing
+      break;
+    default:
+      throw 'Methinks thou hast forgotten something…'
```
Comment (will-in-wi) [inline devtools_m1/cypress/integration/Wishlists/steps.js]: Thanks for retaining my silliness.

File: devtools_m1/cypress/integration/Wishlists/steps.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      cy.get('input[class="input-text qty validate-not-negative-number"]').type('{selectall}0')
+      cy.get('button[title="Update Wishlist"]').first().click()
+      break;
+    case 'the trashcan':
+      cy.get('a[title="Remove Item"]').click()
+      break;
+    default:
+      throw 'Methinks thou hast forgotten something…'
+  }
+})
```
Comment (will-in-wi) [inline devtools_m1/cypress/integration/Wishlists/steps.js]: Nit: Could we have a newline at the end of the file?

Comment (tdieterich-drip) [reply devtools_m1/cypress/integration/Wishlists/steps.js]: Damn you VSCode for not maintaining my IntelliJ setting.

File: devtools_m1/cypress/integration/Wishlists/steps.js (hunk 2 above)
Comment (tdieterich-drip) [reply devtools_m1/cypress/integration/Wishlists/steps.js]: I will do that in a separate PR I think.

**Pull request merged**